tasks/libs/common/check_tools_version.py

```diff
@@ -45,27 +48,41 @@ def custom_golangci_v(v: str) -> str:
     return f"{v}-custom-gcl"
 
 
-def current_golangci_lint_v(ctx: Context) -> str:
+def current_golangci_lint_v(ctx: Context, debug: bool = False) -> str:
     """
     Returns the current user golangci-lint version by running golangci-lint --version
     """
-    cmd = "golangci-lint --version"
-    return ctx.run(cmd, hide=True).stdout.split(' ')[3]
+    debug_flag = "--debug" if debug else ""
+    cmd = f"golangci-lint version {debug_flag}"
+    version_output = ctx.run(cmd, hide=True).stdout
+    return version_output if debug else version_output.split(' ')[3]
 
 
-def check_tools_version(ctx: Context, tools_list: list[str]) -> bool:
+def check_tools_version(ctx: Context, tools_list: list[str], debug: bool = False) -> bool:
     """
     Check that each installed tool in tools_list is the version expected for the repo.
     """
-    is_expected_versions = True
+    should_exit = False
     tools_versions = {
-        'go': {'current_v': current_go_v(ctx), 'expected_v': expected_go_repo_v()},
+        'go': {
+            'current_v': current_go_v(ctx),
+            'expected_v': expected_go_repo_v(),
+            'debug': current_go_v(ctx),
```

Comment:
Should we make it empty if `--debug` is not passed? To have the same behavior between `golangci-lint` and `go`

Reply:
Good point, fixed with [7139880](https://github.com/DataDog/datadog-agent/pull/29802/commits/7139880450e426b6600152bb98c6d4c181af4f2a).